### PR TITLE
[MRG] Update change log in preparation for releasing v0.9.0

### PR DESCRIPTION
**Pull request by betatim**
ref #658 

cc @choldgraf

### CODE REVIEW

**Comment by choldgraf** [comment]
@betatim are we gonna hold off for #662 or not? WDYT?

**Comment by betatim** [comment]
I'd merge this independently of #662 as that PR isn't merged yet it can take care of adding to the change log. Here I tried to tied up after our past sloppiness of not adding to the change log straight away.

**Comment by choldgraf** [comment]
ok - I think I'll give the new release a shot tomorrow if I have a moment, for now I need to go to a graduation ceremony (it's that time of year)

File: docs/source/changelog.rst [hunk 1]
```diff
@@ -34,9 +37,14 @@ Bug fixes
...
   in :pr:`628` by :user:`betatim`.
+- Update to version 5.7.8 of the `notebook` package and version 2.0.12 of
+  `nteract-on-jupyter` in :pr:`650` by :user:`betatim`.
 - Switch to newer version of jupyter-server-proxy to fix websocket handling
   in :pr:`646` by :user:`betatim`.
-- Update to pip v19.0.3 in :pr:`647` by :user:`betatim`.
+- Update to pip version 19.0.3 in :pr:`647` by :user:`betatim`.
+- Ensure ENTRYPOINT is an absolute path in :pr:`657` by :user:`yuvipanda`.
+- Fix handling of `--build-memory-limit` values without a postfix in :pr:`652`
+  by :user:`betatim`.
```
Comment (choldgraf) [inline docs/source/changelog.rst]:
```suggestion
- Create a page naming and describing the "Reproducible Execution
  Environment Specification" (the specification used by repo2docker)
  in :pr:`656` by :user:`choldgraf`.

```

Comment (choldgraf) [reply docs/source/changelog.rst]: shall we add this in and then release?

Comment (betatim) [reply docs/source/changelog.rst]: Ah, this references the issue and not the PR. Let's add this to the change log in #662 itself. Should also be in the new features section, not "bug fix".

Comment (betatim) [reply docs/source/changelog.rst]: -> I removed it again after pressing commit.

Comment (betatim) [reply docs/source/changelog.rst]: Me again ... after having just merged #662 I added your sentence back :D.

**Pull request merged**